bdsx/bds/command.ts

```diff
@@ -622,9 +676,48 @@ export class CommandEnum<V extends string|number|symbol> extends CommandParamete
         }
 
         const registry = serverInstance.minecraft.getCommands().getRegistry();
+        const exists = registry.hasEnum(this.name);
         const enumId = registry.addEnumValues(this.name, [...this.mapper.keys()]);
+        const cmdenum = registry.getEnum(this.name)!;
+
+        const isCxxStringParser = cmdenum.isCxxStringParser();
+        const isIntParser = cmdenum.isIntParser();
+
+        if (!exists) cmdenum.parser = enumParser;
+        // Provide additional parsing on top of the original parser
+        const original = makefunc.js(cmdenum.parser, bool_t, null, CommandRegistry, MantleClass.ref(), StaticPointer, CommandOrigin.ref(), int32_t, CxxString, CxxVector.make(CxxString));
+        cmdenum.parser = makefunc.np((registry, storage, tokenPtr, origin, version, error, errorParams) => {
+            const ret = original(registry, storage, tokenPtr, origin, version, error, errorParams);
+            const token = tokenPtr.getPointerAs(CommandRegistry.ParseToken);
+            const text = token.getText().toLocaleLowerCase();
+            const mapped = this.mapper.get(text);
+            if (mapped != null) {
+                this.results.push(mapped);
+            } else { // Not on the map, it is defined before
+                let result: V;
+                if (isIntParser) {
+                    result = storage.getInt32() as V;
+                } else if (isCxxStringParser) {
+                    result = storage.getCxxString() as V;
+                } else {
+                    // in this case, the value is parsed to some NativeClass, follow the type of the custom enum
+                    if (this instanceof CommandStringEnum) {
+                        result = text as V;
+                    } else {
+                        const values = registry.getEnumValues(this.name)!;
+                        result = values.indexOf(text) as V;
+                    }
+                }
+                this.results.push(result);
```

Comment:
it occurs memory leaks.
it will push the value per command.
but never deleting it.

`CommandEnum` class is not generated per command.
it's generated globally.

Reply:
it seems I missed some other parts.

Reply:
it seems it hooks per enum which has the same name.
and it adds the result per hooks but removes per the command.

Reply:
it seems the previous enum parser makes complicated it.
I will modify it.